corehq/apps/app_manager/suite_xml.py

```diff
@@ -1943,6 +1944,44 @@ def get_target_module(case_type, module_id, with_product_details=False):
             except IndexError:
                 pass
 
+        self.add_parent_datums(datums, module)
+
+        return datums, assertions
+
+    def add_parent_datums(self, datums, module):
+
+        def update_refs(action_, datum_, changed_ids):
+            if action_:
+                # Only advanced module actions have a parent_tag attribute. Basic modules don't need one
+                # because they only deal with one case type. Set parent_tag to None for them.
+                parent_tag = getattr(action_, 'parent_tag', None)
+                if parent_tag in changed_ids:
+                    # update any reference to previously changed datums
+                    change = changed_ids[parent_tag]
+                    nodeset = datum_.nodeset
+                    old = session_var(change['old_id'])
+                    new = session_var(change['new_id'])
+                    datum_.nodeset = nodeset.replace(old, new)
+
+        def avoid_duplicate_ids(action_, this_datum_, parent_datum_, datum_ids_, changed_ids):
+            if action_:
+                case_tag = getattr(action_, 'case_tag', None)  # Set case_tag to None for basic modules
+                if parent_datum_.id in datum_ids_:
+                    # We can't set a new ID to an ID we already have. We need to rename current ID
+                    datum = datum_ids_[parent_datum_.id]
+                    new_id = '_'.join((parent_datum_.id, datum['case_type']))
+                    i = 0
+                    while new_id in datum_ids_:
+                        new_id = '_'.join((parent_datum_.id, datum['case_type'], str(i)))
+                        i += 1
```

Comment:
For normal forms it definitely isn't necessary. It is possible with Advanced Forms but I don't think you need to worry about it since datums for advanced forms are named using the action tags already.